logfire/_internal/integrations/llm_providers/openai.py

```diff
@@ -57,12 +58,14 @@ def get_endpoint_config(options: FinalRequestOptions) -> EndpointConfig:
         return EndpointConfig(
             message_template='Responses API with {gen_ai.request.model!r}',
             span_data={
+                'request_data': json_data,
```

Comment:
This is essentially part of the 'old' format, I don't think we want it here (at least not in full) since it will duplicate the input messages here and in `events`.

Reply:
oh, i'm guessing you need the request/response_data pair in the UI. can you use request_data when it's streaming, `events` otherwise? or, maybe slightly more complicated, see how `events` is handled currently in the non-streaming case.

Reply:
don't even include `events` in the first place when it's streaming

Reply:
I think I follow you now.

I was missing that you could check in `get_endpoint_config` if the request was streaming. I had looked at `FinalRequestOptions` wasn't there so had assumed it wasn't possible, but just found it in `json_data`, so assume your intent was to check that and build `span_data` conditionally, which is what I've now done.

Also made `request_data` conditional, based on your earlier comment but let me know if you want it always present for consistency.